dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java

```diff
@@ -451,14 +451,15 @@ public synchronized void unexport() {
     private void doExportUrls() {
         List<URL> registryURLs = loadRegistries(true);
         for (ProtocolConfig protocolConfig : protocols) {
-            String pathKey = URL.buildKey(getContextPath(protocolConfig).map(p -> p + "/" + path).orElse(path), group, version);
+            URL url = buildExportUrlsFor1Protocol(protocolConfig, registryURLs);

```

Comment:
is the change made here for the problem you mentioned:

> Also found and fixed a bug: when contextpath is set and starts with /，providerModel.getServiceName() does not match with invoker.getUrl().getPathKey()

Reply:
yes, this is because this code:
https://github.com/apache/dubbo/blob/21397b30599ce765d3db938ca53ba26c108ab363/dubbo-common/src/main/java/org/apache/dubbo/common/URL.java#L180-L182

thus `URL.buildKey(getContextPath(protocolConfig).map(p -> p + "/" + path).orElse(path), group, version)` wouldn't match with `url.getPathKey()`